packages/trpc/server/routers/publicViewer/noShow.handler.ts

```diff
@@ -31,33 +45,67 @@ export const noShowHandler = async ({ input }: NoShowOptions) => {
           email: true,
         },
       });
-
+      const allAttendeesMap = allAttendees.reduce((acc, attendee) => {
+        acc[attendee.email] = attendee;
+        return acc;
+      }, {} as Record<string, { id: number; email: string }>);
       const updatePromises = attendees.map((attendee) => {
-        const attendeeToUpdate = allAttendees.find((a) => a.email === attendee.email);
-
-        if (attendeeToUpdate) {
-          return prisma.attendee.update({
-            where: { id: attendeeToUpdate.id },
-            data: { noShow: attendee.noShow },
-          });
-        }
+        const attendeeToUpdate = allAttendeesMap[attendee.email];
+        if (!attendeeToUpdate) return;
+        return prisma.attendee.update({
+          where: { id: attendeeToUpdate.id },
+          data: { noShow: attendee.noShow },
+        });
       });
-
-      await Promise.all(updatePromises);
-    } else {
-      await prisma.booking.update({
-        where: {
-          uid: bookingUid,
-        },
-        data: {
-          noShowHost: true,
+      const results = await Promise.allSettled(updatePromises);
+      const _attendees = results
+        .filter((x) => x.status === "fulfilled")
+        .map((x) => (x as PromiseFulfilledResult<{ noShow: boolean; email: string }>).value)
+        .map((x) => ({ email: x.email, noShow: x.noShow }));
+      const payload = await getResultPayload(_attendees);
+      const booking = await prisma.booking.findUnique({
+        where: { uid: bookingUid },
+        select: {
+          eventType: {
+            select: {
+              id: true,
+              teamId: true,
+              userId: true,
+            },
+          },
         },
       });
+      const orgId = await getOrgIdFromMemberOrTeamId({
+        memberId: booking?.eventType?.userId,
+        teamId: booking?.eventType?.teamId,
+      });
+      const webhooks = await new WebhookService({
+        teamId: booking?.eventType?.teamId,
+        userId: booking?.eventType?.userId,
+        eventTypeId: booking?.eventType?.id,
+        orgId,
+        triggerEvent: WebhookTriggerEvents.BOOKING_NO_SHOW_UPDATED,
+      });
+      // @ts-expect-error payload is too booking specific, we need to refactor this
+      await webhooks.sendPayload({ ...payload, bookingUid });
+      return payload;
     }
+    await prisma.booking.update({
+      where: {
+        uid: bookingUid,
+      },
+      data: {
+        noShowHost: true,
+      },
+    });
+    return { message: "No-show status updated", noShowHost: true };
   } catch (error) {
+    let message = "Failed to update no-show status";
     if (error instanceof Error) {
       logger.error(error.message);
+      message = error.message;
```

Comment:
![image](https://github.com/calcom/cal.com/assets/1780212/ce19cbd8-6b71-42e7-b221-af1f096ee127)
I see the query in the notification in browser, could be due to this change.

Also, it wasn't shown as an error.

Reply:
@hariombalhara have you run migration locally? It seems like prisma is not recognizing the new WebhookTriggerEvents.BOOKING_NO_SHOW_UPDATED eventTrigger.

Reply:
Yeah, that's intentionally done to test the error.

I was concerned that prisma error msg shouldn't be shown to user

Reply:
What would you suggest we do here?

Reply:
I think we should not send the caught error as is and instead we can show a generic error like 'Something went wrong' and log the actual error.

Reply:
Also I think this doesn't happen on production build. Can we verify that?

Reply:
> I think we should not send the caught error as is and instead we can show a generic error like 'Something went wrong' and log the actual error.

done!